src/gui/editors/AutomationEditor.cpp

```diff
@@ -740,91 +738,85 @@ void AutomationEditor::mouseMoveEvent(QMouseEvent * mouseEvent )
 		return;
 	}
 
+	// If the mouse y position is inside the Automation Editor viewport
 	if( mouseEvent->y() > TOP_MARGIN )
 	{
 		float level = getLevel( mouseEvent->y() );
-		int x = mouseEvent->x();
+		// Get the viewport X position where the mouse is at
+		int x = mouseEvent->x() - VALUES_WIDTH;
 
-		x -= VALUES_WIDTH;
+		// If we are moving a value, we account for the offset from the node we
+		// had when clicking it in the first place (user might not have clicked exactly
+		// in the middle of the node)
 		if( m_action == MOVE_VALUE )
 		{
 			x -= m_moveXOffset;
 		}
 
-		int pos_ticks = x * MidiTime::ticksPerBar() / m_ppb +
-							m_currentPosition;
-		// m_mouseDownLeft used to prevent drag when drawing line
+		// Get the X position in ticks
+		int posTicks = (x * MidiTime::ticksPerBar()/m_ppb) + m_currentPosition;
+
+		// If we are in DRAW mode and the left mouse is down
 		if (m_mouseDownLeft && m_editMode == DRAW)
 		{
 			if( m_action == MOVE_VALUE )
 			{
-				// moving value
-				if( pos_ticks < 0 )
+				// If we moved the mouse past the beginning correct the position in ticks
+				if( posTicks < 0 )
 				{
-					pos_ticks = 0;
+					posTicks = 0;
 				}
 
-				drawLine( m_drawLastTick, m_drawLastLevel,
-							pos_ticks, level );
-
-				m_drawLastTick = pos_ticks;
+				m_drawLastTick = posTicks;
 				m_drawLastLevel = level;
 
-				// we moved the value so the value has to be
-				// moved properly according to new starting-
-				// time in the time map of pattern
-				m_pattern->setDragValue( MidiTime( pos_ticks ),
+				// Updates the drag value of the moved node
+				m_pattern->setDragValue( MidiTime( posTicks ),
 								level, true,
 							mouseEvent->modifiers() &
 								Qt::ControlModifier );
-			}
-
-			Engine::getSong()->setModified();
 
+				Engine::getSong()->setModified();
+			}
+			else if( m_action == DRAW_LINE )
+			{
+				// We are drawing a line. For now do nothing (as before), but later logic
+				// could be added here so the line is updated according to the new mouse position
+				// until the button is released
+			}
 		}
 		else if( ( mouseEvent->buttons() & Qt::RightButton &&
 						m_editMode == DRAW ) ||
 				( mouseEvent->buttons() & Qt::LeftButton &&
 						m_editMode == ERASE ) )
 		{
-			// removing automation point
-			if( pos_ticks < 0 )
+			// Removing automation nodes
```

Comment:
We can probably replace the `mouseEvent->buttons()` with the `m_mouseDownX` variables?